src/Migrations/M032MigrateCampaignCoversToP4ColumnsBlock.php

```diff
@@ -0,0 +1,203 @@
+<?php
+
+declare(strict_types=1);
+
+namespace P4\MasterTheme\Migrations;
+
+use P4\MasterTheme\MigrationRecord;
+use P4\MasterTheme\MigrationScript;
+use P4\MasterTheme\BlockReportSearch\BlockSearch;
+use WP_Block_Parser;
+
+/**
+ * Migrate Campaign covers block to Planet4 columns block.
+ */
+class M032MigrateCampaignCoversToP4ColumnsBlock extends MigrationScript
+{
+    private const BLOCK_NAME = 'planet4-blocks/covers';
+    private const POST_TYPES = [ 'page', 'post', 'action', 'campaign' ];
+    private const COVER_TYPE = 'campaign';
+    private const OLD_COVER_TYPES = [
+        '1' => 'take-action',
+        '2' => 'campaign',
+        '3' => 'content',
+    ];
+
+    /**
+     * Extract campaign covers block from page/posts and transform it into Planet4 columns block.
+     *
+     * @param MigrationRecord $record Information on the execution, can be used to add logs.
+     * phpcs:disable SlevomatCodingStandard.Functions.UnusedParameter -- interface implementation
+     */
+    public static function execute(MigrationRecord $record): void
+    {
+        $parser = new WP_Block_Parser();
+
+        // Get the list of posts using Covers block.
+        $posts = self::get_posts_using_block(self::BLOCK_NAME);
+
+        // If there are no posts, abort.
+        if (!$posts) {
+            return;
+        }
+
+        foreach ($posts as $post) {
+            if (empty($post->post_content)) {
+                continue;
+            }
+
+            echo 'Parsing post ', $post->ID, "\n"; // phpcs:ignore
+            $result = false;
+
+            // Go through the post blocks to find the planet4-blocks/covers one.
+            $blocks = $parser->parse($post->post_content);
+
+            foreach ($blocks as &$block) {
+                // For older cover blocks, the cover type is empty, so use the default content cover type,hence skip it.
+                if (!isset($block['attrs']['cover_type'])) {
+                    continue;
+                }
+
+                // For old cover type(earlier it was numeric).
+                if (is_numeric($block['attrs']['cover_type'])) {
+                    $block['attrs']['cover_type'] = self::OLD_COVER_TYPES[ $block['attrs']['cover_type'] ];
+                }
+
+                // Skip other blocks & non campaign cover blocks.
+                if (
+                    ! isset($block['blockName']) || $block['blockName'] !== self::BLOCK_NAME ||
+                    $block['attrs']['cover_type'] !== self::COVER_TYPE
+                ) {
+                    continue;
+                }
```

Comment:
I think these guard clauses should be re-ordered.
IMO, we should first check whether the block name is right, then check the cover type, and finally update the numeric value, something like:

```php
if (!isset($block['blockName']) || $block['blockName'] !== self::BLOCK_NAME) {
             continue;
}
if (!isset($block['attrs']['cover_type']) || $block['attrs']['cover_type'] !== self::COVER_TYPE) {
       continue;
}
if (is_numeric($block['attrs']['cover_type'])) {
         $block['attrs']['cover_type'] = self::OLD_COVER_TYPES[ $block['attrs']['cover_type'] ];
}
```

Reply:
I agree on moving the Block name check to top, but cover type check need to keep below numeric cover type check, cause else we will skip all numeric cover type blocks. I adjusted code as below -

`
                // Skip non cover blocks.
                if (!isset($block['blockName']) || $block['blockName'] !== self::BLOCK_NAME) {
                    continue;
                }

                // For older cover blocks, the cover type is empty, so use the default content cover type, hence skip it.
                if (!isset($block['attrs']['cover_type'])) {
                    continue;
                }

                // For old cover type(earlier it was numeric).
                if (is_numeric($block['attrs']['cover_type'])) {
                    $block['attrs']['cover_type'] = self::OLD_COVER_TYPES[ $block['attrs']['cover_type'] ];
                }

                // Skip non campaign cover blocks.
                if ($block['attrs']['cover_type'] !== self::COVER_TYPE) {
                    continue;
                }
`